DashAISoftware/DashAI
# Feat/uninstall plugin

# Summary


This branch allows the user to uninstall one plugin. This includes removing it from the Python packages directory (by using pip uninstall) and the components registry. This implies modifying the patch plugin API to make it a bit more general because before these changes it just works to update the status param to "installed".


## Changes

- Plugin statuses are reduced to just two possible ones: Registered or Installed
- Tests related to the deleted status "Downloaded" were removed
- Uninstalling a plugin uninstalls it from the Python env (pip uninstall ...)
- Uninstalling a plugin removes it from the component registry: added methods to unregister component classes and also remove them from the relationship manager.


## How to Test


To install and uninstall a plugin it is suggested to see branch [feat/empty-container-for-plugins](https://github.com/DashAISoftware/DashAI/tree/feat/empty-container-for-plugins). This runs DashAI with no components registered from the backend on it. Then go to the "Plugins" tab, press the "refresh" button and install dashai-test2-tabular-classification-package
